src/main/java/me/karakelley/http/ServerConfiguration.java

```diff
@@ -1,55 +1,66 @@
 package me.karakelley.http;
 
+import me.karakelley.http.utility.ArgumentParser;
+import me.karakelley.http.utility.Exit;
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 
+import java.util.Map;
+
 public class ServerConfiguration {
-  private final String[] args;
+  private final Logger logger = LoggerFactory.getLogger(ServerConfiguration.class);
   private final int DEFAULT_PORT = 0;
-  private Integer port = -1;
-  final Logger logger = LoggerFactory.getLogger(ServerConfiguration.class);
+  private static final String DEFAULT_PATH = "nodirectory";
+  private final ArgumentParser parser;
+  private final Exit systemExit;
+  private Integer port;
+  private String directory = "";
+  Map<String, String> argsHash;
 
-  public ServerConfiguration(String[] args) {
-    this.args = args;
+  public ServerConfiguration(ArgumentParser parser, Exit systemExit) {
```

Comment:
Why does the server configuration have to know about command line arguments? I almost feel like the `ServerConfiguration` should just be a simple DTO which is built up from various sources.

I could also see an argument (no pun intended) for the `ServerConfiguration` being an interface and having one particular implementation of that being `CommandLineArguments implements ServerConfiguration` and then in tests you could just create simple implementations of `ServerConfiguration` like...


```
ServerConfiguration config = new ServerConfiguration() {
  @Override
  public int getPort() {
    return 0;
  }

  // ...
}
```